haystack/backends/elasticsearch_backend.py

```diff
@@ -166,6 +176,10 @@ def setup(self):
         )
         current_mapping = {"modelresult": {"properties": field_mapping}}
 
+        explicit_dynamic_setting = haystack.connections[self.connection_alias].options.get('DYNAMIC', None)
```

Comment:
Instead of creating the index the immediately having to CURL the mapping API to disable dynamic mapping this lets us control the dynamic setting via settings files.